test/Handler/SitesSpec.hs

```diff
@@ -108,3 +111,9 @@ spec = withApp $ do
 
 createSite_ :: UserId -> Site -> DB ()
 createSite_ userId = void . createSite userId
+
+selectLanguage :: Text -> RequestBuilder m ()
+selectLanguage lang = do
```

Comment:
> this could be a good pull request to Yesod.Test down the line

Definitely. This Friday, I'm going to work on https://github.com/yesodweb/yesod/issues/956, but moving this (and everything else in our Yesod.Test.Extension) to a PR could be good to do next Friday.